flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/DispatcherCleanupITCase.java

```diff
@@ -288,7 +304,7 @@ public void testCleanupAfterLeadershipChange() throws Exception {
         dispatcherGateway.submitJob(jobGraph, TIMEOUT).get();
 
         waitForJobToFinish(leaderElectionService, dispatcherGateway, jobId);
-        jobGraphRemovalErrorReceived.await();
+        firstCleanupFailsLatch.trigger();
```

Comment:
I don't see where we guarantee that the cleanup isn't re-run immediately and finishes before the leadership has changed.

Reply:
The Dispatcher is configured to not retry. But I also added another assert that checks that cleanup is only called once after the leadership is lost. 👍

Reply:
btw. this trigger call was the reason for the flakyness of the test. 🤦 I swapped trigger and await again. We want to wait here for the cleanup and not trigger it.

Reply:
> The Dispatcher is configured to not retry

Where is that done?